marshal.go

```diff
@@ -979,12 +979,12 @@ func marshalVarbind(pdu *SnmpPDU) ([]byte, error) {
 
 // -- Unmarshalling Logic ------------------------------------------------------
 
-func (x *GoSNMP) unmarshalHeader(packet []byte, response *SnmpPacket) (int, error) {
+func (x *GoSNMP) unmarshalVersionFromHeader(packet []byte, response *SnmpPacket) (int, SnmpVersion, error) {
```

Comment:
Wouldn't it make more sense to have the cursor second and version first? (as getting the version is the main goal of this method)
```suggestion
func (x *GoSNMP) unmarshalVersionFromHeader(packet []byte, response *SnmpPacket) (SnmpVersion, int, error) {
```

Reply:
good call out, will amend this :-)